Разметка всех страниц

TaliUltra opened the pull request:
Разметка первой страницы и страницы каталога

---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss-individual/1/user/2252007/tasks/4)

TaliUltra committed:
Разметка - первая страница

TaliUltra committed:
Разметка страницы каталога

TaliUltra committed:
Update Readme.md

Komnatnaya commented on Readme.md:
```diff
@@ -1,7 +1,7 @@
 
 - Студент: [Nataliya Malikova](https://up.htmlacademy.ru/htmlcss-individual/1/user/2252007).
-- Наставник: `Екатерина Комнатная` (https://htmlacademy.ru/profile/id215761).
+- Наставник: [Екатерина Комнатная] (https://htmlacademy.ru/profile/id215761).
```
Пробел между скобками `](` лишний

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,352 @@
+  <body>
+    <header class="main-header">
+      <!-- User navigation -->
+      <img class="logo-black" src="" alt="Логотип фирмы Девайс." />
```
Логотип - это ссылка на главную. И если на главной он может быть просто картинкой, то на внутренних страницах это обязательно ссылка.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+          method="post"
+        >
+          <input
+            type="image"
```
Это не кнопка отправки формы, а просто декоративный элемент, поэтому `input type="image"` тут не нужен.

Komnatnaya replied:
Не исправлено, этот инпут стоит удалить.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+          />
+          <label class="visually-hidden" for="search-device">Device</label>
+          <input
+            type="text"
```
Неверный тип поля поиска.

TaliUltra replied:
А какой тогда?  Пользователь вводит текст - название товара, чтобы его найти.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+            id="search-device"
+            required
+          />
+          <button name="search" class="button-search" type="submit">
```
Кнопке отправки атрибут name не нужен, потому что серверу не нужно информация с нее.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+        <ul class="user-navigation-list">
+          <li class="user-navigation-item">
+            <a class="user-navigation-link" href="#">
+              <img src="" alt="Вход в личный кабинет." />Войти
```
Иконки - не контентные изображения, а декор, поэтому добавлять их в разметку тегом img не нужно.

TaliUltra replied:
А как добавлять не контентные изображения?

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+      <!-- Main navigation -->
+      <nav class="main-navigation">
+        <p class="items-catalog">Каталог товаров</p>
+        <img src="" alt="Каталог товаров." />
```
Плюсик не контентное изображение, к томе же он меняет свой внешний вид при взаимодействии с пользователем.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
 
+      <!-- Main navigation -->
+      <nav class="main-navigation">
+        <p class="items-catalog">Каталог товаров</p>
```
Это пункт меню на каталог товаров, т.е. это ссылка

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+        <h2 class="visually-hidden">Примеры товаров</h2>
+
+        <button class="slider-button slider-prev" type="button">
+          <img src="" alt="Предыдущий товар." />
```
Картинки в кнопка - не контентные, тег img не нужен.

Komnatnaya replied:
А вот подпись к кнопкам без текста нужна, ее можно положить в скрытый спан

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+
+        <button class="slider-button slider-next" type="button">
+          <img src="" alt="Следующий товар." />
+        </button>
```
Кнопки "Предыдущий товар" и "Следующий товар." - это кнопки переключения слайдера. Картинка, текстовая часть и кнопка "Подробнее" - это один слайд слайдера. Всего их три, можно посмотреть в стайлгайде. 
Кнопки переключения слайдера не должны быть внутри слайда, они размечаются под слайдами.
Т.е. эти кнопки стоит положить рядом со списком slider-pagination

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+          <img src="" alt="Следующий товар." />
+        </button>
+
+        <img src="" alt="Номер один." />
```
Это не картинка, а текст.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+
+        <ul class="services-list">
+          <li class="services-item">
+            <a class="services-link services-link-current" href="#">Доставка</a>
```
Блок услуг - это тоже своеобразный слайдер, называет табы.
Т.е. кнопки services-link переключают блок справа (их тоже можно посмотреть в стайлгайде).
Поскольку кнопки никуда не ведут, их стоит сделать button

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+          совершенно бесплатно! Ведь мы неплохо заработаем, поднимая его на ваш
+          этаж.
+        </p>
+        <img src="" alt="Быстрая доставка товаров." class="delivery-img" />
```
Не контентное изображение

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+        <div class="delivery-rare-item">
+          <a href="#" class="delivery-rare-item-link">
+            <img
+              src=""
```
Не контентное изображение

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+              class="delivery-img"
+            />
+            <p>Доставим редкий товар под заказ</p>
+            <img src="" alt="Информация о доставке." class="info" />
```
Не контентное изображение

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+
+      <!-- About us -->
+      <section class="about-us-contacts">
+        <h2 class="visually-hidden">О нас</h2>
```
Два раза повторяется заголовок "О нас". Я бы советовала заменить section class="about-us-contacts" на div.
А div class="about-us" и div class="contacts" на  section, тогда озаглавливать about-us-contacts не придется

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+          </p>
+          <ul class="delivery-companies">
+            <li class="delivery-companies-item">
+              <img src="" alt="" />Меловые линии
```
Декор, даже не картинка. Такие элементы делаются стилями.

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+        </form>
+
+        <div class="free">
+          <img src="" alt="" />
```
Не контентное изображение

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,381 @@
+          <ul class="social-list">
+            <li class="social-item">
+              <a class="social-link" href="https://t.me/htmlacademy">
+                <img src="" alt="Иконка социальной сети Телеграм." />
```
Не контентное изображение.

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,352 @@
+        <form
+          class="search-form"
+          action="https://echo.htmlacademy.ru/"
+          method="post"
```
Метод get, мы же хотим получить результат поиска.

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,352 @@
+    <main class="main-content">
+      <h1>Каталог моноподов</h1>
+      <section class="page-content">
+        <h2>Моноподы для селфи</h2>
```
h1. Не стоит придумывать заголовки, если есть видимые в каталоге.

Komnatnaya replied:
Не исправлено.
Правка о том что "Моноподы для селфи" - это заголовок первого уровня.
На каждой странице должен быть заголовок первого уровня и это как раз он.

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,352 @@
+        <ul class="breadcrubs">
+          <li class="breadcrubs-item">
+            <a href="#">Главная</a>
+            <img src="" alt="Указатель." />
```
Не контентное изображение

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,352 @@
+              <li class="catalog-filter-item">
+                <label for="" class="control-label">
+                  <input
+                    name="color-black"
```
У радио должны быть одинаковые name и разные value, иначе они будут отправлять на сервер одно и тоже, независимо от выбора пользователя

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,352 @@
+        <ul class="sort-list">
+          <li class="sort-item">
+            <a href="#">
+              <img src="" alt="Вверх." />
```
Не контентные изображения

TaliUltra committed:
Вносит изменения которые поняла

TaliUltra commented:
Внесла изменения, которые поняла.
На счет контентного изображения у меня были вопросы при разметке.
+ еще кучка мелочных вопросов.
Есть вопросы по слайдерам: как их оформлять.
Можем встрериться обсудить?

TaliUltra committed:
правки

1. Не уверена в правильности списка карточек с кнопкой 'подробнее'.
2. Index.html line 175. Как быть с slider-pagination-button-current для остальных двух товаров?

Komnatnaya commented on index.html:
```diff
@@ -1,12 +1,446 @@
+              <img src="" alt="Квадрокоптер с мощным лазером." />
+            </a>
+
+            <a href="#" class="samples-item-one">
```
Это просто текст, ссылка тут не нужна.

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,370 @@
+                class="product-card-img"
+              />
+            </a>
+            <button type="button">Подробнее</button>
```
Подробнее - это ссылка на карточку товара.

Komnatnaya commented on catalog.html:
```diff
@@ -0,0 +1,370 @@
+                alt="Любительская селфи-палка."
+                class="product-card-img"
+              />
+            </a>
```
Картинку ссылкой можно не оборачивать, потому что ее все равно накроет серый фон при наведении мыши и  появится кнопка "подробнее"

TaliUltra committed:
разметка правка 3
